scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/SortedBucketSink.java

```diff
@@ -343,21 +339,26 @@ public void processElement(ProcessContext c) {
       final V record = c.element();
       final K key = extractKeyFn.apply(record);
 
-      final KV<BucketShardId, byte[]> bucketIdAndSortBytes =
-          Optional.ofNullable(getResource().getIfPresent(key))
```

Comment:
The `key` param to `getIfPresent` is marked `@NonNull`, how does that work before when `key == null`?  I guess the entire `Optional` becomes `None` and it hits `orElseGet` then `processKey`, which should still handle the null key?

Reply:
I think it was a bug before... we just didn't have a test case for it.